local build env: golangci-link related issues causing make failure

first time trying to get a local build env setup.   

first: 

`make ensure` runs cleanly, even without `golangci-lint` installed.    that should probably be addressed.   I'll also open a PR to add this to the `brew install`...

next:

brew installed with `brew install golangci/tap/golangci-lint`
and got:
`golangci-lint has version 1.17.1 built from 4ba2155 on 2019-06-10T09:07:35Z`

when running `make` I get multiple linting failures (clean master at `4833fe1045cad8440705505bc6bf32b3a9b2ba28`:

```
LINT:
golangci-lint run
cmd/pulumi-language-nodejs/main.go:87:56: `langauge` is a misspelling of `language` (misspell)
	cmdutil.InitTracing("pulumi-language-nodejs", "pulumi-langauge-nodejs", tracing)
	                                                      ^
make[1]: *** [lint] Error 1
make: *** [sdk/nodejs_default] Error 2
```

if I fix that one, it moves on to more typos:

```
cheese:~/git/github/pulumi/pulumi(master)%% make lint
LINT:
golangci-lint run
pkg/resource/plugin/provider_plugin_test.go:106:62: `properites` is a misspelling of `properties` (misspell)
		assert.Truef(t, reflect.DeepEqual(fromVal, val), "expected properites %s to be deeply equal", key)
		                                                           ^
tests/integration/integration_test.go:226:54: `sould` is a misspelling of `could` (misspell)
	contract.AssertNoErrorf(err, "resource.NewUniqueHex sould not fail with no maximum length is set")
	                                                    ^
tests/integration/integration_test.go:268:54: `sould` is a misspelling of `could` (misspell)
	contract.AssertNoErrorf(err, "resource.NewUniqueHex sould not fail with no maximum length is set")
	                                                    ^
make: *** [lint] Error 1
```


comparing that to a recent travis build - https://travis-ci.com/pulumi/pulumi/builds/115992790

which just does:

```LINT:
golangci-lint run
./node_modules/.bin/tslint -c tslint.json -p tsconfig.json
TEST ALL:```

the travis build doesnt mention (that I saw) what version of `golangci-lint` is being run - so there could be a version issue.   (I haven't dug into try to see if it's defined somewhere else)

otherwise - I'd suspect there's a config file somewhere that's missing from the repo that defines the linting rules?

--- a/pkg/resource/plugin/provider_plugin_test.go
+++ b/pkg/resource/plugin/provider_plugin_test.go
@@ -103,7 +103,7 @@ func TestAnnotateSecretsDifferentProperties(t *testing.T) {
 			continue
 		}
 
-		assert.Truef(t, reflect.DeepEqual(fromVal, val), "expected properites %s to be deeply equal", key)
+		assert.Truef(t, reflect.DeepEqual(fromVal, val), "expected properties %s to be deeply equal", key)
 	}
 
 	_, has := to["extraFromValue"]

--- a/sdk/nodejs/cmd/pulumi-language-nodejs/main.go
+++ b/sdk/nodejs/cmd/pulumi-language-nodejs/main.go
@@ -84,7 +84,7 @@ func main() {
 
 	args := flag.Args()
 	logging.InitLogging(false, 0, false)
-	cmdutil.InitTracing("pulumi-language-nodejs", "pulumi-langauge-nodejs", tracing)
+	cmdutil.InitTracing("pulumi-language-nodejs", "pulumi-language-nodejs", tracing)
 
 	nodePath, err := exec.LookPath("node")
 	if err != nil {

--- a/sdk/proto/generate.sh
+++ b/sdk/proto/generate.sh
@@ -7,7 +7,7 @@
 # hence error prone.
 #
 # This script relies only on Docker. The container holds the installation of gRPC, tools, etc., for
-# different langauges, so nothing is else required to be installed on your machine.
+# different languages, so nothing is else required to be installed on your machine.
 set -e
 
 # First build our Protobuf/gRPC compiler Docker image, so dev machines don't need it.

--- a/tests/integration/integration_test.go
+++ b/tests/integration/integration_test.go
@@ -223,7 +223,7 @@ func TestRemoveWithResourcesBlocked(t *testing.T) {
 	}()
 
 	stackName, err := resource.NewUniqueHex("rm-test-", 8, -1)
-	contract.AssertNoErrorf(err, "resource.NewUniqueHex sould not fail with no maximum length is set")
+	contract.AssertNoErrorf(err, "resource.NewUniqueHex should not fail with no maximum length is set")
 
 	e.ImportDirectory("single_resource")
 	e.RunCommand("pulumi", "stack", "init", stackName)
@@ -265,7 +265,7 @@ func TestPreviewJSON(t *testing.T) {
 	}
 
 	stackName, err := resource.NewUniqueHex("rm-test-", 8, -1)
-	contract.AssertNoErrorf(err, "resource.NewUniqueHex sould not fail with no maximum length is set")
+	contract.AssertNoErrorf(err, "resource.NewUniqueHex should not fail with no maximum length is set")
 
 	e.ImportDirectory("single_resource")
 	e.RunCommand("pulumi", "stack", "init", stackName)

